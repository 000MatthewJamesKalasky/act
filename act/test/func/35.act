template<pint N>
deftype str (int<N> x, z)
{
   methods {
     macro plus (int<2> y) : int<4>
     {
       self := x + y
     }
   }
}

defproc test()
{
   str<4> s;
   int<8> w;
   chp {
      w := 0x42;
      s := str(0x42);
      log ("got ", s.x, " ", s.z)
   }
}

test t;
