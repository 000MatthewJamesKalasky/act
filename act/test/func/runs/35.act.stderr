ERROR: File `35.act', line 18, col 9
	Typechecking failed on CHP assignment
  stmt: s := str(66)
	Incompatible structures
